codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -524,31 +322,13 @@ void window(int initialWindowSize) {
         }
 
         @Override
-        void allocate(int bytes) {
-            allocated += bytes;
-            // Also artificially reduce the streamable bytes for this tree to give the appearance
-            // that the data has been written. This will be restored before the allocated bytes are
-            // actually written.
-            incrementStreamableBytesForTree(-bytes);
-        }
-
-        @Override
-        int writeAllocatedBytes() {
-            int numBytes = allocated;
-
-            // Restore the number of streamable bytes to this branch.
-            incrementStreamableBytesForTree(allocated);
-            resetAllocated();
-
-            // Perform the write.
-            return writeBytes(numBytes);
-        }
-
-        /**
-         * Reset the number of bytes that have been allocated to this stream by the priority algorithm.
-         */
-        private void resetAllocated() {
-            allocated = 0;
+        int writeAssignedBytes(int assigned) {
+            try {
+                // Perform the write.
+                return writeBytes(assigned);
+            } finally {
+                streamByteAssigner.update(this);
```

Comment:
It doesn't seem like `streamableBytes` will necessarily change either. `updateStreamableBytes` sounds better.